test/sql/catalog/function/test_table_macro_complex.test

```diff
@@ -0,0 +1,41 @@
+# name: test/sql/catalog/function/test_table_macro_complex.test
+# description: Test SELECT Macro
+# group: [function]
+
+statement ok
```

Comment:
Looking at the code coverage, there are still a number of missing tests, particularly w.r.t. storage of table macros. The serialize method is never being run. Could you add 
a test similar to `test/sql/storage/catalog/test_macro_storage.test` for table macro functions?